src/main/java/net/neoforged/neoforge/client/entity/animation/json/AnimationLoader.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation.json;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.mojang.logging.LogUtils;
+import java.util.HashMap;
+import java.util.Map;
+import net.minecraft.client.animation.AnimationDefinition;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+/**
+ * A loader for entity animations written in JSON. You can also get parsed animations from this class.
+ */
+public final class AnimationLoader extends SimpleJsonResourceReloadListener {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public static final AnimationLoader INSTANCE = new AnimationLoader();
+
+    private final Map<ResourceLocation, AnimationHolder> animations = new HashMap<>();
```

Comment:
I notice animations are never cleared from here, so the holders will stay around if the player unloads a resource pack using them. Probably not very significant though since it's just the `ResourceLocation`. If it is I think it could be solved by making the values weakly referenced, then adding an additional list to keep strong references only to the actually loaded holders.